packages/augur-sdk/src/api/Gnosis.ts

```diff
@@ -1,77 +1,401 @@
+
+    return (
+      '0x' +
+      ethUtil
+        .generateAddress2(
+          this.augur.contracts.proxyFactory.address,
+          salt,
+          initCode
+        )
+        .toString('hex')
+    );
+  }
+
   async getGnosisSafeAddress(account: string): Promise<string> {
     return this.augur.contracts.gnosisSafeRegistry.getSafe_(account);
   }
 
   async createGnosisSafeDirectlyWithETH(account: string): Promise<string> {
-    const gnosisSafeRegistryAddress = this.augur.contracts.gnosisSafeRegistry.address;
-    const cashAddress = this.augur.contracts.cash.address;
-    const shareTokenAddress = this.augur.contracts.shareToken.address;
-    const augurAddress = this.augur.contracts.augur.address;
-    const createOrderAddress = this.augur.contracts.createOrder.address;
-    const fillOrderAddress = this.augur.contracts.fillOrder.address;
-    const registrationData = await this.provider.encodeContractFunction("GnosisSafeRegistry", "callRegister", [gnosisSafeRegistryAddress, augurAddress, createOrderAddress, fillOrderAddress, cashAddress, shareTokenAddress]);
-    const gnosisSafeData = await this.provider.encodeContractFunction("GnosisSafe", "setup", [[account], 1, gnosisSafeRegistryAddress, registrationData, NULL_ADDRESS, NULL_ADDRESS, 0, NULL_ADDRESS]);
+    const gnosisSafeData = await this.buildGnosisSetupData(account);
+
     // Make transaction to proxy factory
-    const nonce = Date.now();
-    const proxy = this.augur.contracts.proxyFactory.createProxyWithNonce_(this.augur.contracts.gnosisSafe.address, gnosisSafeData, new BigNumber(nonce));
-    await this.augur.contracts.proxyFactory.createProxyWithNonce(this.augur.contracts.gnosisSafe.address, gnosisSafeData, new BigNumber(nonce));
+    const nonce = AUGUR_GNOSIS_SAFE_NONCE;
+    const proxy = this.augur.contracts.proxyFactory.createProxyWithNonce_(
+      this.augur.contracts.gnosisSafe.address,
+      gnosisSafeData,
+      new BigNumber(nonce)
+    );
+    await this.augur.contracts.proxyFactory.createProxyWithNonce(
+      this.augur.contracts.gnosisSafe.address,
+      gnosisSafeData,
+      new BigNumber(nonce)
+    );
     return proxy;
   }
 
-  async createGnosisSafeViaRelay(params: GetGnosisSafeAddressParams): Promise<string> {
-    const gnosisSafeRegistryAddress = this.augur.contracts.gnosisSafeRegistry.address;
+  async createGnosisSafeViaRelay(
+    params: CreateGnosisSafeViaRelayParams
+  ): Promise<SafeResponse> {
+    if (this.gnosisRelay === undefined) {
+      throw new Error('No Gnosis Relay provided to Augur SDK');
+    }
+
+    const gnosisSafeRegistryAddress = this.augur.contracts.gnosisSafeRegistry
+      .address;
+
+    const setupData = await this.buildRegistrationData();
+
+    try {
+      const response = await this.gnosisRelay.createSafe({
+        saltNonce: AUGUR_GNOSIS_SAFE_NONCE,
+        owners: [params.owner],
+        threshold: 1,
+        paymentToken: params.paymentToken,
+        to: gnosisSafeRegistryAddress,
```

Comment:
Are we passing the `to` but not the `data` because that part isnt working yet?

Reply:
@justinbarry? 
Are we going to circle back on this? Can you create an issue to track the work needed here?

Reply:
I added it as `setupData` because that is how the relay referred to it. 🤷‍♂